app/Http/Requests/Api/RoleRequest.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+namespace App\Http\Requests\Api;
+
+use Illuminate\Foundation\Http\FormRequest;
+
+class RoleRequest extends FormRequest
+{
+    /**
+     * Determine if the user is authorized to make this request.
+     */
+    public function authorize(): bool
+    {
+        return true;
+    }
+
+    /**
+     * Get the validation rules that apply to the request.
+     *
+     * @return array<string, \Illuminate\Contracts\Validation\ValidationRule|array|string>
+     */
+    public function rules(): array
+    {
+        return [
+            'name' => 'required|string|unique:roles,name',
+            'guard_name' => 'required|string',
```

Comment:
what is guard name here?

Reply:
As far as I can see in the Laravel roles documentation `guard-name` is an authentication provider associated with each role. when we create the new role then the laravel gives 2 guards by default one is `web` and the other is `api`. So we can define multiple guards based on application requirements. like as `web-admin` and `api-user`, `api-admin`.

Reply:
Why do we need a guard name in this file? Are you sending this value in the API payload? Why haven't you put guard name in other FormRequest files?

Reply:
Yes, I'm sending the `guard_name` value in this API payload only which will be `web` by default. Because we send the value of `guard_name` in this payload only while creating the role and the `guard_name` column exists only in the role table. That's why I have added `guard_name` only in this `FormRequest` file.